README.md

```diff
@@ -64,19 +69,159 @@ const main = defineCommand({
+
+// Then, you define a main command and attach sub command to it
+const main = defineCommand({
+  meta: {
+    name: "hello",
+    version: "1.0.0",
+    description: "My Awesome CLI App",
+  },
+  commands: {
+    sub, // Attach sub command to main command
+  },
+});
+
+runMain(main);
+```
+
+### Hooks
+
+`citty` supports a `setup` and `cleanup`  functions that are called before and after command execution. This is useful for setting up and cleaning up resources.
+
+Only the `setup` and `cleanup` functions from the command called are executed. For example, if you run `hello sub`, only the `setup` and `cleanup` functions from `sub` command are executed and not the ones from `hello` command.
+
+```js
+import { defineCommand, runMain } from "citty";
+
+const main = defineCommand({
+  meta: {
+    name: "hello",
+    version: "1.0.0",
+    description: "My Awesome CLI App",
+  },
+  setup() { // Setup function is called before command execution or before any sub command execution
+    console.log("Setting up...");
+  },
+  cleanup() { // Cleanup function is called after command execution or after any sub command execution
+    console.log("Cleaning up...");
+  },
+  run() {
+    console.log("Hello World!");
+  },
+});
+
+runMain(main);
+```
+
+### Lazy Load Commands
+
+For large CLI apps, you may want to only load the command that is being executed.
+
+First, create a command in a file and export it.
+
+```js
+import { defineCommand } from "citty";
+
+export default defineCommand({
+  meta: {
+    name: "sub",
+    description: "Sub command",
+  },
+  run({ args }) {
+    console.log(`Hello ${args.name}!`);
+  },
+});
+```
+
+Then, create the main command and import the sub command.
+
+```js
+const main = defineCommand({
+  meta: {
+    name: "hello",
+    version: "1.0.0",
+    description: "My Awesome CLI App",
+  },
+  commands: {
+    sub: () => import("./sub.js").then((m) => m.default), // Lazy Import Sub Command
+  },
+});
+```
+
+Now, when you run `hello sub`, the sub command will be loaded and executed. This avoid to load all commands at once.
+
+### Publish CLI App as an Executable
+
+You must first bundle your CLI app. To do so, you can use [`unjs/unbuild`](https://github.com/unjs/unbuild).
+
+Then, you must create a file named `index.mjs` in a folder named `bin` at the root of your package. This file must export the main command from the `dist` build.
+
+```js
+#!/usr/bin/env node
+
+import { runMain } from '../dist/index.mjs'
+
+runMain()
+```
+
+Then, you will need to update your `package.json` file to enable the usage as a CLI:
+
+```json
+{
+  "type": "module",
+  "bin": "./bin/index.mjs",
+  // Name of the CLI will be the name of the package. You can provide an object to change the name.
+  // @see https://docs.npmjs.com/cli/v10/configuring-npm/package-json#bin
+  // "bin": {
+  //   "my-cli": "./bin/index.mjs"
+  // },
+  "files": [
+    "bin",
+    "dist"
+  ]
+}
+```
+
+You're ready to publish your CLI app to npm!
+
 ## Utils
```

Comment:
Should we rename it to API and add more stuff like what other "type" are available for commands and explain default & required?

Reply:
Do you refer to the `args` object @jd-solanki from `defineCommand`?

Reply:
Yes `args`'s `type` property.

Reply:
Yes, good idea! I will add a section about it.

Reply:
Added!